shotover/src/transforms/kafka/sink_single.rs

```diff
@@ -106,7 +98,11 @@ impl Transform for KafkaSinkSingle {
     async fn transform<'a>(&'a mut self, mut requests_wrapper: Wrapper<'a>) -> Result<Messages> {
         if self.outbound.is_none() {
             let codec = KafkaCodecBuilder::new(Direction::Sink);
-            let tcp_stream = tcp::tcp_stream(self.connect_timeout, &self.address).await?;
+            let tcp_stream = tcp::tcp_stream(
+                self.connect_timeout,
+                (requests_wrapper.local_addr.ip(), self.address_port),
```

Comment:
How does this works in case a machine has several IPs and/or interfaces?

Reply:
In the case where shotover and kafka have seperate NICs/IPs it would not have worked in the first place since we only rewrite the IP.
If you need that kind of setup then I believe you will need to use KafkaSinkCluster instead of KafkaSinkSingle.